### DISCUSSION TITLE
Use FF variables in rules

### DISCUSSION THREAD

**Discussion by Gsyltc** [firefly-iii/firefly-iii#7157] [Firefly III]
Hello,

I wanted to know if it was possible to use variables in firefly rules.
Example 1 
Trigger: "The amount is greater than"
Trigger on value: theaccount.balance() (where theaccount.balance() is the function of returns the account balance)

In the example, we could say that if the value of the transaction is higher than the balance of the account, the transaction is cancelled because it is incompatible with the rules of the account

Thanks

**Comment by JC5** [comment]
This isn't possible, see also [the documentation](https://docs.firefly-iii.org/firefly-iii/more-information/what-its-not/#more-complicated-rules-like-boolean-logic-copypasting-values-etc).

What you want can be done with the [API](https://api-docs.firefly-iii.org/) or with [webhooks](https://docs.firefly-iii.org/firefly-iii/features/webhooks/) however. For that though, you're on your own.

**Comment by Gsyltc** [comment]
Hi James,

Thanks. Yes, It will be possible with the API. It was jsute a question.

Have a nice day
